module/VuFind/src/VuFind/Http/GuzzleService.php

```diff
@@ -96,13 +94,114 @@ public function __construct(array $config)
      * @param ?string $url     Target URL (required for proper proxy setup for non-local addresses)
      * @param ?float  $timeout Request timeout in seconds (overrides configuration)
      *
-     * @return \GuzzleHttp\ClientInterface
+     * @return ClientInterface
      */
-    public function createClient(?string $url = null, ?float $timeout = null): \GuzzleHttp\ClientInterface
+    public function createClient(?string $url = null, ?float $timeout = null): \Psr\Http\Client\ClientInterface
     {
         return new \GuzzleHttp\Client($this->getGuzzleConfig($url, $timeout));
     }
 
+    /**
+     * Perform a GET request.
+     *
+     * @param string $url     Request URL
+     * @param array  $params  Request parameters (query string)
+     * @param float  $timeout Request timeout in seconds
+     * @param array  $headers Request HTTP headers
+     *
+     * @return ResponseInterface
+     */
+    public function get(
+        string $url,
+        array $params = [],
+        ?float $timeout = null,
+        array $headers = []
+    ): ResponseInterface {
+        if ($params) {
+            $query = $this->createQueryString($params);
+            if (str_contains($url, '?')) {
+                $url .= '&' . $query;
+            } else {
+                $url .= '?' . $query;
+            }
+        }
+        $client = $this->createClient($url, $timeout);
+        $options = [];
+
+        if ($headers) {
+            $options['headers'] = $headers;
+        }
+
+        return $client->request('GET', $url, $options);
+    }
+
+    /**
+     * Perform a POST request.
+     *
+     * @param string $url     Request URL
+     * @param mixed  $body    Request body document
+     * @param string $type    Request body content type
+     * @param float  $timeout Request timeout in seconds
+     * @param array  $headers Request HTTP headers
+     *
+     * @return ResponseInterface
+     */
+    public function post(
+        string $url,
+        $body = null,
+        string $type = 'application/octet-stream',
+        ?float $timeout = null,
+        array $headers = []
+    ): ResponseInterface {
+        $client = $this->createClient($url, $timeout);
+
+        $options = [
+            'body' => $body,
+            'headers' => array_merge(
+                [
+                    'Content-Type' => $type,
+                    'Content-Length' => strlen($body ?? ''),
+                ],
+                $headers
+            ),
+        ];
+
+        return $client->request('POST', $url, $options);
+    }
+
+    /**
+     * Create a query string from an array of parameters.
+     *
+     * @param array $params Parameters
```

Comment:
More specific comment?

```suggestion
     * @param array $params Parameters (either an associative key=>value array, or a regular array of preformatted key=value strings)
```